tools/src/tester/StoryEvaluator.ts

```diff
@@ -0,0 +1,78 @@
+import { type Chapter, type Story, type SupplementalChapter } from './types/story.types'
+import { type ChapterEvaluation, Result, type StoryEvaluation } from './types/eval.types'
+import ChapterEvaluator from './ChapterEvaluator'
+
+export interface StoryFile {
+  display_path: string
+  full_path: string
+  story: Story
+}
+
+export default class StoryEvaluator {
+  story: Story
+  display_path: string
+  full_path: string
+  skipped: boolean = false
+  result: Result = Result.PASSED
+
+  constructor (story_file: StoryFile) {
+    this.story = story_file.story
+    this.display_path = story_file.display_path
+    this.full_path = story_file.full_path
+  }
+
+  async evaluate (): Promise<StoryEvaluation> {
+    if (this.story.skipped) {
+      return {
+        result: Result.SKIPPED,
+        display_path: this.display_path,
+        full_path: this.full_path,
+        description: this.story.description,
+        chapters: []
+      }
+    }
+    return {
+      display_path: this.display_path,
+      full_path: this.full_path,
+      description: this.story.description,
+      prologues: await this.#evaluate_supplemental_chapters(this.story.prologues),
+      chapters: await this.#evaluate_chapters(this.story.chapters),
+      epilogues: await this.#evaluate_supplemental_chapters(this.story.epilogues),
+      result: this.result
+    }
+  }
+
+  async #evaluate_chapters (chapters: Chapter[]): Promise<ChapterEvaluation[]> {
+    let skipped: boolean = this.skipped
```

Comment:
I think this should be `has_errors` rather than `skipped` to be clearer about purpose